File: Cap 07/7.4 - Uso de Matrizes Dinamicas/matdin.go
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 *
 * Capitulo ..: 7 - Topico 7.4
 * 
 * Programa ..: MATRIZ DINAMICA
 */

package main

import (
  "fmt"
)

var N int
var A []string

func main() {

  fmt.Println("MATRIZ DINAMICA")
  fmt.Println()

  fmt.Print("Entre a quantidade de elementos da matriz: ")
  fmt.Scanln(&N)
  fmt.Println()

  // Ajuste da matriz para quantidade solicitada
  A := make([]string, N)

  for I := 0; I <= (N - 1); I++ {
    fmt.Printf("Entre o %3do. nome: ", I + 1)
    fmt.Scanln(&A[I])
  }

  fmt.Println()
  fmt.Println("Foram fornecidos os nomes:")
  fmt.Println()

  for I := 0; I <= (N - 1); I++ {
    fmt.Printf("Nome %3d = %s\n", I + 1, A[I])
  }

  // Nao e necessario liberar a memoria alocada
 
}

File: Cap 07/7.4 - Uso de Matrizes Dinamicas/matdin.php
<?php
/*
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 7 - Topico 7.4
 * 
 * Programa ..: MATRIZ DINAMICA
 */

echo "MATRIZ DINAMICA\n\n";

echo "Entre a quantidade de elementos da matriz: ";
$N = readline();
echo "\n";
  
// Ajuste da matriz para quantidade solicitada
$A = array();

for ($i = 1; $i <= $N; $i++) {
    echo "Entre o " . str_pad($i, 3, ' ', STR_PAD_LEFT) . "o. nome: ";
    $nome = readline();
    $A[$i] = $nome;
}

echo "\n";
echo "Foram fornecidos os nomes:\n\n";

for ($i = 1; $i <= $N; $i++) {
    echo "Nome " . str_pad($i, 3, ' ', STR_PAD_LEFT) . " = " . $A[$i] . "\n";
}

// Nao e necessario liberar a memoria alocada

?>
